studio/components/interfaces/SQLEditor/UtilityPanel/ReadOnlyBadge.tsx

```diff
@@ -0,0 +1,16 @@
+import { Badge } from 'ui'
+import { useSqlEditorStateSnapshot } from 'state/sql-editor'
+import { useUser } from 'common'
+
+export type ReadOnlyBadgeProps = { id: string }
+const ReadOnlyBadge = ({ id }: ReadOnlyBadgeProps) => {
+  const user = useUser()
+  const snap = useSqlEditorStateSnapshot()
+
+  const snippet = snap.snippets[id]
```

Comment:
I'm getting `Application error: a client-side exception has occurred (see the browser console for more information).`. The console says `TypeError: s is undefined` where `s` is probably `snippet` uglified.